### ISSUE TITLE
Zfs fails to build with autoconf 2.70

### ISSUE THREAD

**Issue by edacval** [openzfs/zfs#11413]
### System information
Type | Version/Name
 --- | ---
Distribution Name	| Arch Linux
Distribution Version	| latest
Linux Kernel	| any
Architecture	| x64
ZFS Version	| 0.8.5, 2.0.0

### Describe the problem you're observing
After update autoconf to  [2.70 ](https://lists.gnu.org/archive/html/autotools-announce/2020-12/msg00001.html) version, ./autogen.sh exits with error `configure.ac: error: AM_GNU_GETTEXT is used, but not AM_GNU_GETTEXT_VERSION or AM_GNU_GETTEXT_REQUIRE_VERSION`.

### Describe how to reproduce the problem
```
git checkout zfs-0.8.5
#   same error with zfs-2.0.0 
./autogen.sh
```
### Include any warning/errors/backtraces from the system logs
./autogen.sh output:
```
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force -I config
configure.ac:39: warning: The macro `AC_CANONICAL_SYSTEM' is obsolete.
configure.ac:39: You should run autoupdate.
./lib/autoconf/general.m4:2080: AC_CANONICAL_SYSTEM is expanded from...
configure.ac:39: the top level
configure.ac:50: warning: The macro `AC_PROG_LIBTOOL' is obsolete.
configure.ac:50: You should run autoupdate.
config/libtool.m4:99: AC_PROG_LIBTOOL is expanded from...
configure.ac:50: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
./lib/autoconf/general.m4:1533: AC_ARG_ENABLE is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/always-python.m4:8: ZFS_AC_CONFIG_ALWAYS_PYTHON is expanded from...
config/zfs-build.m4:155: ZFS_AC_CONFIG_ALWAYS is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/always-pyzfs.m4:23: ZFS_AC_CONFIG_ALWAYS_PYZFS is expanded from...
config/zfs-build.m4:155: ZFS_AC_CONFIG_ALWAYS is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/mount-helper.m4:1: ZFS_AC_CONFIG_USER_MOUNT_HELPER is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/user-udev.m4:1: ZFS_AC_CONFIG_USER_UDEV is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/user-systemd.m4:1: ZFS_AC_CONFIG_USER_SYSTEMD is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/user-sysvinit.m4:1: ZFS_AC_CONFIG_USER_SYSVINIT is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/user-dracut.m4:1: ZFS_AC_CONFIG_USER_DRACUT is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:2846: AC_TRY_COMPILE is expanded from...
config/user-makedev.m4:4: ZFS_AC_CONFIG_USER_MAKEDEV_IN_SYSMACROS is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:2846: AC_TRY_COMPILE is expanded from...
config/user-makedev.m4:24: ZFS_AC_CONFIG_USER_MAKEDEV_IN_MKDEV is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/user-libexec.m4:1: ZFS_AC_CONFIG_USER_ZFSEXEC is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:39: warning: The macro `AC_CANONICAL_SYSTEM' is obsolete.
configure.ac:39: You should run autoupdate.
./lib/autoconf/general.m4:2080: AC_CANONICAL_SYSTEM is expanded from...
configure.ac:39: the top level
configure.ac:50: warning: The macro `AC_PROG_LIBTOOL' is obsolete.
configure.ac:50: You should run autoupdate.
config/libtool.m4:99: AC_PROG_LIBTOOL is expanded from...
configure.ac:50: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
./lib/autoconf/general.m4:1533: AC_ARG_ENABLE is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/always-python.m4:8: ZFS_AC_CONFIG_ALWAYS_PYTHON is expanded from...
config/zfs-build.m4:155: ZFS_AC_CONFIG_ALWAYS is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/always-pyzfs.m4:23: ZFS_AC_CONFIG_ALWAYS_PYZFS is expanded from...
config/zfs-build.m4:155: ZFS_AC_CONFIG_ALWAYS is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/mount-helper.m4:1: ZFS_AC_CONFIG_USER_MOUNT_HELPER is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/user-udev.m4:1: ZFS_AC_CONFIG_USER_UDEV is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/user-systemd.m4:1: ZFS_AC_CONFIG_USER_SYSTEMD is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/user-sysvinit.m4:1: ZFS_AC_CONFIG_USER_SYSVINIT is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/user-dracut.m4:1: ZFS_AC_CONFIG_USER_DRACUT is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:2846: AC_TRY_COMPILE is expanded from...
config/user-makedev.m4:4: ZFS_AC_CONFIG_USER_MAKEDEV_IN_SYSMACROS is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:2846: AC_TRY_COMPILE is expanded from...
config/user-makedev.m4:24: ZFS_AC_CONFIG_USER_MAKEDEV_IN_MKDEV is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/user-libexec.m4:1: ZFS_AC_CONFIG_USER_ZFSEXEC is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
autoreconf: configure.ac: tracing
configure.ac: error: AM_GNU_GETTEXT is used, but not AM_GNU_GETTEXT_VERSION or AM_GNU_GETTEXT_REQUIRE_VERSION
autoreconf: running: libtoolize --copy --force
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, 'config'.
libtoolize: copying file 'config/ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'config'.
libtoolize: copying file 'config/libtool.m4'
libtoolize: copying file 'config/ltoptions.m4'
libtoolize: copying file 'config/ltsugar.m4'
libtoolize: copying file 'config/ltversion.m4'
libtoolize: copying file 'config/lt~obsolete.m4'
autoreconf: configure.ac: not using Intltool
autoreconf: configure.ac: not using Gtkdoc
autoreconf: running: aclocal --force -I config
configure.ac:39: warning: The macro `AC_CANONICAL_SYSTEM' is obsolete.
configure.ac:39: You should run autoupdate.
./lib/autoconf/general.m4:2080: AC_CANONICAL_SYSTEM is expanded from...
configure.ac:39: the top level
configure.ac:50: warning: The macro `AC_PROG_LIBTOOL' is obsolete.
configure.ac:50: You should run autoupdate.
config/libtool.m4:99: AC_PROG_LIBTOOL is expanded from...
configure.ac:50: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
./lib/autoconf/general.m4:1533: AC_ARG_ENABLE is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/always-python.m4:8: ZFS_AC_CONFIG_ALWAYS_PYTHON is expanded from...
config/zfs-build.m4:155: ZFS_AC_CONFIG_ALWAYS is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/always-pyzfs.m4:23: ZFS_AC_CONFIG_ALWAYS_PYZFS is expanded from...
config/zfs-build.m4:155: ZFS_AC_CONFIG_ALWAYS is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/mount-helper.m4:1: ZFS_AC_CONFIG_USER_MOUNT_HELPER is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/user-udev.m4:1: ZFS_AC_CONFIG_USER_UDEV is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/user-systemd.m4:1: ZFS_AC_CONFIG_USER_SYSTEMD is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/user-sysvinit.m4:1: ZFS_AC_CONFIG_USER_SYSVINIT is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/user-dracut.m4:1: ZFS_AC_CONFIG_USER_DRACUT is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:2846: AC_TRY_COMPILE is expanded from...
config/user-makedev.m4:4: ZFS_AC_CONFIG_USER_MAKEDEV_IN_SYSMACROS is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:2846: AC_TRY_COMPILE is expanded from...
config/user-makedev.m4:24: ZFS_AC_CONFIG_USER_MAKEDEV_IN_MKDEV is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/user-libexec.m4:1: ZFS_AC_CONFIG_USER_ZFSEXEC is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:39: warning: The macro `AC_CANONICAL_SYSTEM' is obsolete.
configure.ac:39: You should run autoupdate.
./lib/autoconf/general.m4:2080: AC_CANONICAL_SYSTEM is expanded from...
configure.ac:39: the top level
configure.ac:50: warning: The macro `AC_PROG_LIBTOOL' is obsolete.
configure.ac:50: You should run autoupdate.
config/libtool.m4:99: AC_PROG_LIBTOOL is expanded from...
configure.ac:50: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
./lib/autoconf/general.m4:1533: AC_ARG_ENABLE is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/always-python.m4:8: ZFS_AC_CONFIG_ALWAYS_PYTHON is expanded from...
config/zfs-build.m4:155: ZFS_AC_CONFIG_ALWAYS is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/always-pyzfs.m4:23: ZFS_AC_CONFIG_ALWAYS_PYZFS is expanded from...
config/zfs-build.m4:155: ZFS_AC_CONFIG_ALWAYS is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/mount-helper.m4:1: ZFS_AC_CONFIG_USER_MOUNT_HELPER is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/user-udev.m4:1: ZFS_AC_CONFIG_USER_UDEV is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/user-systemd.m4:1: ZFS_AC_CONFIG_USER_SYSTEMD is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/user-sysvinit.m4:1: ZFS_AC_CONFIG_USER_SYSVINIT is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/user-dracut.m4:1: ZFS_AC_CONFIG_USER_DRACUT is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:2846: AC_TRY_COMPILE is expanded from...
config/user-makedev.m4:4: ZFS_AC_CONFIG_USER_MAKEDEV_IN_SYSMACROS is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:2846: AC_TRY_COMPILE is expanded from...
config/user-makedev.m4:24: ZFS_AC_CONFIG_USER_MAKEDEV_IN_MKDEV is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/user-libexec.m4:1: ZFS_AC_CONFIG_USER_ZFSEXEC is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
autoreconf: running: /usr/bin/autoconf --force
configure.ac:39: warning: The macro `AC_CANONICAL_SYSTEM' is obsolete.
configure.ac:39: You should run autoupdate.
./lib/autoconf/general.m4:2080: AC_CANONICAL_SYSTEM is expanded from...
configure.ac:39: the top level
configure.ac:50: warning: The macro `AC_PROG_LIBTOOL' is obsolete.
configure.ac:50: You should run autoupdate.
config/libtool.m4:99: AC_PROG_LIBTOOL is expanded from...
configure.ac:50: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
./lib/autoconf/general.m4:1533: AC_ARG_ENABLE is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/always-python.m4:8: ZFS_AC_CONFIG_ALWAYS_PYTHON is expanded from...
config/zfs-build.m4:155: ZFS_AC_CONFIG_ALWAYS is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/always-pyzfs.m4:23: ZFS_AC_CONFIG_ALWAYS_PYZFS is expanded from...
config/zfs-build.m4:155: ZFS_AC_CONFIG_ALWAYS is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/mount-helper.m4:1: ZFS_AC_CONFIG_USER_MOUNT_HELPER is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/user-udev.m4:1: ZFS_AC_CONFIG_USER_UDEV is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/user-systemd.m4:1: ZFS_AC_CONFIG_USER_SYSTEMD is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/user-sysvinit.m4:1: ZFS_AC_CONFIG_USER_SYSVINIT is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/user-dracut.m4:1: ZFS_AC_CONFIG_USER_DRACUT is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:2846: AC_TRY_COMPILE is expanded from...
config/user-makedev.m4:4: ZFS_AC_CONFIG_USER_MAKEDEV_IN_SYSMACROS is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:2846: AC_TRY_COMPILE is expanded from...
config/user-makedev.m4:24: ZFS_AC_CONFIG_USER_MAKEDEV_IN_MKDEV is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/user-libexec.m4:1: ZFS_AC_CONFIG_USER_ZFSEXEC is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
autoreconf: running: /usr/bin/autoheader --force
autoreconf: running: automake --add-missing --copy --force-missing
configure.ac:39: warning: The macro `AC_CANONICAL_SYSTEM' is obsolete.
configure.ac:39: You should run autoupdate.
./lib/autoconf/general.m4:2080: AC_CANONICAL_SYSTEM is expanded from...
configure.ac:39: the top level
configure.ac:50: warning: The macro `AC_PROG_LIBTOOL' is obsolete.
configure.ac:50: You should run autoupdate.
config/libtool.m4:99: AC_PROG_LIBTOOL is expanded from...
configure.ac:50: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
./lib/autoconf/general.m4:1533: AC_ARG_ENABLE is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/always-python.m4:8: ZFS_AC_CONFIG_ALWAYS_PYTHON is expanded from...
config/zfs-build.m4:155: ZFS_AC_CONFIG_ALWAYS is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/always-pyzfs.m4:23: ZFS_AC_CONFIG_ALWAYS_PYZFS is expanded from...
config/zfs-build.m4:155: ZFS_AC_CONFIG_ALWAYS is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/mount-helper.m4:1: ZFS_AC_CONFIG_USER_MOUNT_HELPER is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/user-udev.m4:1: ZFS_AC_CONFIG_USER_UDEV is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/user-systemd.m4:1: ZFS_AC_CONFIG_USER_SYSTEMD is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/user-sysvinit.m4:1: ZFS_AC_CONFIG_USER_SYSVINIT is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/user-dracut.m4:1: ZFS_AC_CONFIG_USER_DRACUT is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:2846: AC_TRY_COMPILE is expanded from...
config/user-makedev.m4:4: ZFS_AC_CONFIG_USER_MAKEDEV_IN_SYSMACROS is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:2846: AC_TRY_COMPILE is expanded from...
config/user-makedev.m4:24: ZFS_AC_CONFIG_USER_MAKEDEV_IN_MKDEV is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:59: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:59: You should run autoupdate.
./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
config/user-libexec.m4:1: ZFS_AC_CONFIG_USER_ZFSEXEC is expanded from...
config/user.m4:4: ZFS_AC_CONFIG_USER is expanded from...
config/zfs-build.m4:168: ZFS_AC_CONFIG is expanded from...
configure.ac:59: the top level
configure.ac:49: installing 'config/compile'
configure.ac:42: installing 'config/missing'
cmd/mount_zfs/Makefile.am: installing 'config/depcomp'
autoreconf: 'config/config.sub' is updated
autoreconf: 'config/config.guess' is updated
autoreconf: 'config/install-sh' is updated
autoreconf: Leaving directory '.'
```

**Comment by sempervictus** [comment]
Yeah, just ran into this too. `autoupdate` doesnt help here

**Comment by behlendorf** [comment]
Opened PR #11419 to address this.

Note it doesn't resolve the following warning.  However, this warning isn't fatal, isn't specific to 2.70, and isn't harmful.  Additional investigation will be needed to fully quiet this warning with the gettext integration.

```
configure.ac: error: AM_GNU_GETTEXT is used, but not AM_GNU_GETTEXT_VERSION or AM_GNU_GETTEXT_REQUIRE_VERSION.
```

**Comment by eli-schwartz** [comment]
The warning is actually an "error" as specified by the message text, it *is* fatal (autoreconf exits 1), and it *is* (in this incarnation) specific to 2.70...

The autoconf 2.69 output is:

```
autoreconf: configure.ac: AM_GNU_GETTEXT is used, but not AM_GNU_GETTEXT_VERSION
```
(no mention of "error", does not return failure)

...

I noticed this and posted to the autoconf mailing list trying to understand the cause: https://lists.gnu.org/archive/html/autoconf/2020-12/msg00044.html

**Comment by tycho** [comment]
I applied @behlendorf's 2.70 patch and the `autogen.sh` script still exits with status code 1 (presumably due to the `AM_GNU_GETTEXT` related error). Yet it seems to have succeeded because the build goes without a hitch if I just ignore the `autogen.sh` exit code and try building anyway (i.e. `./autogen.sh || true`). This feels like an autoconf bug.

**Comment by behlendorf** [comment]
@eli-schwartz you're right, it does appear to have been promoted to an error.  I overlooked that since as @tycho mentioned even after the error everything can still be build properly.  We'll of course need to get it sorted out.  But while we're looking at there I don't think there's any reason to hold off merging the other unrelated autoconf 2.70 updates.

**Comment by 65a** [comment]
@tycho, ignoring the error (at least for me) allows the compile but the DKMS files are hosed:
PKGBUILD I'm using: https://pastebin.com/pe2DbAYA
```
==> dkms install --no-depmod -m zfs -v 2.0.0rc1.r311.g064c2cf40 -k 5.9.14-arch1-1
configure: error: cannot find required auxiliary files: compile
Error! Bad return status for module build on kernel: 5.9.14-arch1-1 (x86_64)
```
As a workaround, is it safe to remove user-gettext.m4 temporarily?

**Comment by tycho** [comment]
> @tycho, ignoring the error (at least for me) allows the compile but the DKMS files are hosed:

Uh, hm. Seems to work here?

```
(3/5) Arming ConditionNeedsUpdate...
(4/5) Install DKMS modules
==> dkms install --no-depmod -m zfs -v 2.0.0+rc1+r311+g064c2cf40 -k 5.10.4-1-hsw
==> depmod 5.10.4-1-hsw
(5/5) Updating linux initcpios...
```

**Comment by 65a** [comment]
Possibly a packaging related issue, if I figure out the difference I'll post here. FYI @eli-schwartz (I'm just using a version from the zfs-dkms-git AUR + @behlendorf's patch)

I can confirm removing the function contents in user-gettext.m4 prevents the autoconf error.

**Comment by eli-schwartz** [comment]
>  But while we're looking at there I don't think there's any reason to hold off merging the other unrelated autoconf 2.70 updates.

That's true. PR #11419 does not solve this issue, but it does solve the unreported issue "in addition, there are a bunch of noisy deprecation warnings consuming many lines of output". Much appreciated for the PR -- it will make it easier to see only important lines in the output of autoreconf. :)

> I can confirm removing the function contents in user-gettext.m4 prevents the autoconf error.

I'm sure it does -- at the "cost" of removing gettext functionality (and resulting in linker errors if libc does not contain a gettext symbol usable without `-lintl` and `-I/path/to/intl/headers` -- glibc is one that does include gettext directly) from the project, which in the specific case of the dkms module does not matter, since the kernel modules don't link to libc/libintl and invoke gettext(3), nor distribute .mo catalogs.

Admittedly, it does not seem like the utils distribute .mo catalogs either, despite decorating several message strings as translatable (and of course linking to libc/libintl) -- meaning this intl infrastructure is spending time in order to do nothing since there's nothing to translate *to*. Perhaps there was intent at some point to set up a translation team that never panned out?



> `configure: error: cannot find required auxiliary files: compile`

My packaging scripts do not install the entire source tree, to cut down on the size. This file was not needed for autoconf 2.69 and is needed for autoconf 2.70; I would update the PKGBUILD but it's blocked by the gettext problem...

@tycho's PKGBUILD is everything I moved away from; it is 5 times the download size and nearly 3 times the installed size due to the high number of unused files it packages.

**Comment by behlendorf** [comment]
> Perhaps there was intent at some point to set up a translation team that never panned out?

The backstory here is that the ZFS code has always decorated the CLI messages with `gettext()` in order to make it possible to provide translations.  It's possible these were provided for Solaris originally but I'm not sure.  Regardless, we opted to preserve the infrastructure even if we've never in practice shipped any translations.  @rkitover did the original autoconf work to detect `libintl/libiconv` in e8864b1b28c231d80f0df8b1c4b893f9ada73aad and may have some thoughts about the best way to resolve this.  As a workaround we could simply disable it for the moment to resolve the build issues.

**Comment by tycho** [comment]
> @tycho's PKGBUILD is everything I moved away from; it is 5 times the download size and nearly 3 times the installed size due to the high number of unused files it packages.

Looks like most of the package size in my PKGBUILD is just `/usr/share/zfs/zfs-tests`. This cuts it by a little over half:

```diff
diff --git a/zfs-git/PKGBUILD b/zfs-git/PKGBUILD
index b4c64bd5..2115c421 100644
--- a/zfs-git/PKGBUILD
+++ b/zfs-git/PKGBUILD
@@ -35,6 +35,8 @@ prepare() {
   cd "$srcdir/${_module}"
   patch -Np1 -i ../0001-dsl_scan-allow-users-to-clear-scan-logs-entirely.patch
   patch -Np1 -i ../0001-Autoconf-2.70-compatibility.patch
+  sed -ri '/^[ \t]*tests\//d' configure.ac
+  sed -ri 's/ tests / /' Makefile.am
 }

 build() {
@@ -72,6 +74,7 @@ package_zfs-dkms() {
   install -d ${pkgdir}/usr/src
   cp -a ${srcdir}/${_module}-${pkgver} ${pkgdir}/usr/src/
   rm -rf ${pkgdir}/usr/src/${_module}-${pkgver}/.git
+  rm -rf ${pkgdir}/usr/src/${_module}-${pkgver}/tests
 }

 package_zfs-utils() {
```

Anyway, I guess we're getting a bit off topic.

**Comment by risto42** [comment]
wouldn't just adding `AM_GNU_GETTEXT_VERSION([0.19.6])`, or possibly a subsequent version if need be, after `AM_GNU_GETTEXT([external])` in user-gettext.m4 do the job?

**Comment by tycho** [comment]
> wouldn't just adding `AM_GNU_GETTEXT_VERSION([0.19.6])`, or possibly a subsequent version if need be, after `AM_GNU_GETTEXT([external])` in user-gettext.m4 do the job?

Sadly, no, that doesn't solve it. It still spews the same error and returns a nonzero exit code. It seems like autoconf has trouble noticing that `AM_GNU_GETTEXT_VERSION` is present when it and `AM_GNU_GETTEXT` are inside a macro (or maybe nested in a couple of macros?) for some reason.

**Comment by behlendorf** [comment]
Opened PR #11430 with a minimal change to avoid the fatal error and restore the status quo for the moment.

**Comment by eli-schwartz** [comment]
> I noticed this and posted to the autoconf mailing list trying to understand the cause: https://lists.gnu.org/archive/html/autoconf/2020-12/msg00044.html

The fatal error should be fixed by https://git.savannah.gnu.org/cgit/autoconf.git/commit/?h=branch-2.70&id=35b1708c3961dce31e4ce43014e71d77aa4b5e31

See the autoconf maintainer's reply here: https://lists.gnu.org/archive/html/autoconf/2021-01/msg00006.html

> I think what this project actually wants, at least for right now, is to not run autopoint.  To make that possible again, I have demoted the "AM_GNU_GETTEXT is used, but not AM_GNU_GETTEXT_VERSION" message to a warning.  The fix is on development trunk as commit 66fdaea3cfb4e758212c1891913e9a59441d49af, and the 2.70 branch as 35b1708c3961dce31e4ce43014e71d77aa4b5e31.  It will be included in the bug-fix release coming out some time this week.

So, there should be a new bugfix release of autoconf in a few days that no longer errors out here.

**Comment by behlendorf** [comment]
@eli-schwartz that's excellent news!  In that case, there's no need for us to add a workaround and I'll close that PR.

**Comment by behlendorf** [comment]
Closing.  This should be resolved with the next version of autoconf.  The required changes for 2.70 were included in OpenZFS v2.0.1 and upstream autoconf will be switching the AM_GNU_GETTEXT error back to a warning.

**Comment by rkitover** [comment]
@behlendorf if you would like to improve the localization infrastructure, I could help with this.

We would not need to manage our own team of translators, there are services like transifex where people do this and you just pull updates from them, I do this for another project I work on.
